lib/vector-buffers/src/variants/disk_v2/io.rs

```diff
@@ -129,21 +131,21 @@ impl Filesystem for ProductionFilesystem {
     type MutableMemoryMap = memmap2::MmapMut;
 
     async fn open_file_writable(&self, path: &Path) -> io::Result<Self::File> {
-        tokio::fs::OpenOptions::new()
-            .append(true)
-            .read(true)
-            .create(true)
-            .open(path)
-            .await
+        let mut open_options = tokio::fs::OpenOptions::new();
+        open_options.append(true).read(true).create(true);
+
+        configure_file_open_options_for_write(&mut open_options);
+
+        open_options.open(path).await
```

Comment:
This is an undesirable pattern (passing `&mut open_options`) compared to the builder pattern of the original. Since the ordering of the mode option doesn't matter (I think? correct me if I'm wrong), could we wrap the `new` and `mode` bits into the custom function, resulting in something like:
```suggestion
        new_file_with_write_options()
            .append(true)
            .read(true)
            .create(true)
            .open(path)
            .await
```

Reply:
In ba90f4421, I've taken your suggestion and applied it to all areas that were creating some configuration of `OpenOptions`.

It's a little more code changed, but it deduplicates more code and the helper methods are at least documented so they drive home more of the intent, I believe.